test/integration/run_envoy_test.sh

```diff
@@ -22,3 +22,24 @@ expect_fail_with_error "PARSE ERROR: Argument: --bogus-flag" --bogus-flag
 start_test Launching envoy without --config-path or --config-yaml fails.
 expect_fail_with_error \
   "At least one of --config-path and --config-yaml should be non-empty"
+
+start_test Launching envoy with unknown IP address.
```

Comment:
I'm surprised all these tests pass (emit the expected string) without adding custom configurations to trigger those conditions. It makes me question whether expect_fail_with_error (which I wrote) really works.

Reply:
I need some help in adding custom configurations to trigger test conditions. I believe it is done through yaml files in test/config/integration ?

Reply:
Ah these tests don't pass :)  For each of these commands you can probably add --config-yaml "...config that triggers that error...".  Or you can add the configs as files and reference them with "-c filename".

For synthesizing up a config string in bash that triggers that error, see hotrestart_test.sh -- there's some messiness involving ipv6 vs ipv4 which you'll need to understand.

Reply:
I have gone through hotrestart_test.sh, it is mentioned that work has been duplicated from test_environment.cc via sed. I am confused about where to add the config that triggers the error - whether to do it through C++ or shell script. Can you please clarify and point me to some code sample? :)

Reply:
For ex - the following configs should trigger the error
1. envoy -c hello --local-address-ip-version foo
2. envoy --max-obj-name-len 1
3. envoy --max-stats 1000000000

Reply:
My suggestion is: ignore the TODO in hotrestart_test.sh. Instead, just copy the pattern a bit for how it generates a config based on whether the test is running in ipv6 or not.

This is sub-optimal of course because what you are changing has nothing really to do with IPV6, but it's hard to write a config that is agnostic to ipv4/6 and therefore your test needs to know about it. We should fix that, both for C++ tests and sh tests.  Feel free to take that on :)

Or it's OK if you just copy the pattern in hotrestart_test.sh. Remember all you are trying to do is get a mostly-legal config, that you then -- for each subtest -- modify to make illegal so that you can trigger a specific error.

Reply:
https://github.com/envoyproxy/envoy/issues/5556

Reply:
I wrote a test with flag `-c hello` to run envoy and I am facing this issue - 

```
[2019-01-12 10:43:56.182][024611][critical][main] [source/server/server.cc:86] error initializing configuration 'hello': unable to read file: hello
[2019-01-12 10:43:56.182][024611][info][main] [source/server/server.cc:521] exiting
unable to read file: hello

```
The test in invoked at `test/integration/run_envoy_test.sh`. Any suggestions about how to pass config-path to run envoy? Also, is there any default configuration to pass for tests?

Reply:
Yes -- the issue is that when you run tests under bazel, they are 'hermetic' and any files you need other than the shell-script itself must be listed under the 'data' section of the bazel rule. You can find adjacent examples of that data section in the BUILD file, in the rule for hotrestart_test, and you need to follow that pattern of pulling in the config files into the rule for run_envoy_test as well.